src/components/InputandFilterPanel/InputandFilterPanel.tsx

```diff
@@ -1,37 +1,69 @@
+import React, { useState } from "react";
 import {
+	ContinentOption,
+	FilterButton,
 	InputImage,
-	StyledCountriesSelect,
-	StyledCountriesSelectLabel,
+	StyledCustomSelect,
+	StyledFilterPanel,
 	StyledForm,
+	StyledImg,
 	StyledInputElement,
 	StyledInputLabel,
 	StyledInputandFilterSection,
 } from "./InputandFilterPanel.styles";
 import searchIcon from "../../images/search.png";
+import arrowIcon from "../../images/arrow.png";
 
-export const InputandFilterPanel = () => (
-	<StyledInputandFilterSection>
-		<StyledForm>
-			<StyledInputLabel htmlFor="country_search">
-				<InputImage src={searchIcon} alt="" />
-				<StyledInputElement
-					id="country_search"
-					placeholder="Search for a country..."
-				/>
-			</StyledInputLabel>
-		</StyledForm>
+const ContinentsArray = [
+	{ id: 1, name: "Africa" },
+	{ id: 2, name: "America" },
+	{ id: 3, name: "Asia" },
+	{ id: 4, name: "Europe" },
+	{ id: 5, name: "Oceania" },
+];
 
-		<StyledCountriesSelectLabel htmlFor="continents">
-			<StyledCountriesSelect name="continents" id="continents">
-				<option value="" selected disabled hidden>
-					Filter by region
-				</option>
-				<option value="Africa">Africa</option>
-				<option value="America">America</option>
-				<option value="Asia">Asia</option>
-				<option value="Europa">Europa</option>
-				<option value="Oceania">Oceania</option>
-			</StyledCountriesSelect>
-		</StyledCountriesSelectLabel>
-	</StyledInputandFilterSection>
-);
+export const InputandFilterPanel = () => {
+	const [isMenuOpen, setIsMenuOpen] = useState(false);
+	const [selectOption, setSelectOption] = useState("Filter by Region");
+
+	const MenuToggle = () => {
+		setIsMenuOpen((prevState) => !prevState);
+	};
+
+	const SelectOptionsHandle = (e: React.ChangeEvent<HTMLLIElement>) => {
+		setSelectOption(e.target.innerText);
+	};
+
+	return (
+		<StyledInputandFilterSection>
+			<StyledForm>
+				<StyledInputLabel htmlFor="country_search">
+					<InputImage src={searchIcon} alt="" />
+					<StyledInputElement
+						id="country_search"
+						placeholder="Search for a country..."
+					/>
+				</StyledInputLabel>
+			</StyledForm>
+
+			<StyledFilterPanel onClick={MenuToggle}>
+				<span>{selectOption}</span>
+				<FilterButton type="button">
+					<StyledImg
+						src={arrowIcon}
+						alt="button to open continents select"
+					></StyledImg>
+				</FilterButton>
+				{isMenuOpen ? (
```

Comment:
Can't you make isMenuOpen && componentName instead of ternary operator?

Reply:
fixed.